python/pyspark/pandas/indexes/datetimes.py

```diff
@@ -741,6 +741,13 @@ def pandas_at_time(pdf) -> ps.DataFrame[int]:
             psdf = psdf.pandas_on_spark.apply_batch(pandas_at_time)
         return ps.Index(first_series(psdf).rename(self.name))
 
+    def map(
+        self,
+        mapper: Union[dict, Callable[[Any], Any], pd.Series] = None,
+        na_action: Optional[str] = None,
+    ) -> "Index":
+        return MissingPandasLikeDatetimeIndex.map(self, mapper, na_action)
```

Comment:
qq: why can't we support it here?

Reply:
Good question, let me try if inheriting `base` directly works.